File: app/src/main/java/example/login/LoginActivity.java
package example.login;

import android.arch.lifecycle.ViewModelProviders;
import android.content.Intent;
import android.databinding.Observable;
import android.view.View;

import com.style.base.BaseActivity;
import com.style.bean.User;
import com.style.framework.R;
import com.style.framework.databinding.ActivityLoginBinding;

import example.home.MainActivity;

public class LoginActivity extends BaseActivity {

    private long userId = 18;
    private ActivityLoginBinding bd;
    private LoginModel loginModel;

    @Override
    public int getLayoutResId() {
        return R.layout.activity_login;
    }

    @Override
    protected boolean isGeneralTitleBar() {
        return false;
    }

    @Override
    public void initData() {
        bd = getBinding();
        loginModel = ViewModelProviders.of(this).get(LoginModel.class);
        loginModel.loginSucceed.addOnPropertyChangedCallback(new Observable.OnPropertyChangedCallback() {
            @Override
            public void onPropertyChanged(Observable observable, int i) {
                if (loginModel.loginSucceed.get()) {
                    startActivity(new Intent(getContext(), MainActivity.class));
                    finish();
                }
            }
        });
        loginModel.user.addOnPropertyChangedCallback(new Observable.OnPropertyChangedCallback() {
            @Override
            public void onPropertyChanged(Observable observable, int i) {
                setUserView(loginModel.user.get());
            }
        });
        loginModel.loginState.observe(this, aBoolean -> {
            if (aBoolean){
                startActivity(new Intent(getContext(), MainActivity.class));
                finish();
            }
        });
        loginModel.login();
    }

    public void login(View v) {
        String userId = bd.etAccount.getText().toString();
        String password = bd.etPassword.getText().toString();
        //mPresenter.login(userId, password);
    }

    public void setUserView(User user) {
        bd.etAccount.setText(user.userName);
        bd.etPassword.setText(user.password);
    }

    public void loginSuccess() {

    }

    public void loginFailed() {

    }
}

File: app/src/main/java/example/login/LoginActivity.kt
package example.login

import android.content.Intent
import android.os.Bundle
import android.text.method.DigitsKeyListener
import android.view.View
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProvider
import com.style.base.BaseActivity
import com.style.entity.UserInfo
import com.style.framework.R
import com.style.framework.databinding.ActivityLoginBinding
import example.home.MainActivity

class LoginActivity : BaseActivity() {

    private val userId: Long = 18
    private lateinit var bd: ActivityLoginBinding
    private lateinit var loginModel: LoginModel

    override fun onCreate(arg0: Bundle?) {
        super.onCreate(arg0)
        bd = ActivityLoginBinding.inflate(layoutInflater)
        setContentView(bd.root)
        setFullScreenStableDarkMode(false)

        val digits = getContext().getString(R.string.digits_password)
        bd.etPassword.keyListener = DigitsKeyListener.getInstance(digits)
        //keyListener与inputType有冲突，先设置的会被覆盖
        //bd.etPassword.inputType = InputType.TYPE_CLASS_TEXT
        loginModel = ViewModelProvider(this).get(LoginModel::class.java)

        loginModel.user.observe(this, object : Observer<UserInfo> {
            override fun onChanged(t: UserInfo?) {

                setUserView(loginModel.user.value!!)
            }
        })
        loginModel.loginState.observe(this, Observer<Boolean> { t ->
            if (t!!) {
                startActivity(Intent(getContext(), MainActivity::class.java))
                finish()
            } else {

            }
        })

        bd.btSignIn.setOnClickListener {}
        bd.btSignIn.setOnClickListener(object : View.OnClickListener {
            override fun onClick(v: View) {
                login()
            }
        })
        bd.btSignIn.setOnClickListener { view ->
            view.visibility = View.VISIBLE
            view.visibility = View.VISIBLE
        }
        login()
    }

    fun login() {
        val userId = bd.etAccount.text.toString()
        val password = bd.etPassword.text.toString()
        //mPresenter.login(userId, password);
        loginModel.login()
    }

    fun setUserView(user: UserInfo) {
        bd.etAccount.setText(user.userName)
        bd.etPassword.setText(user.password)
    }

    fun loginSuccess() {

    }

    fun loginFailed() {

    }
}